### PR TITLE
Implemented PartialEq and Eq for Image and SubImage

### PR DESCRIPTION
**Pull request by za-songguo**
Thank you for helping out with embedded-graphics development! Please:

- [x] Check that you've added passing tests and documentation
- [ ] Add a `CHANGELOG.md` entry in the **Unreleased** section under the appropriate heading (**Added**, **Fixed**, etc) if your changes affect the **public API**
- [x] Run `rustfmt` on the project
- [x] Run `just build` (Linux/macOS only) and make sure it passes. If you use Windows, check that CI passes once you've opened the PR.

## PR description

Implemented `PartialEq` and `Eq` for Image and `SubImage`.

### CODE REVIEW

File: benches/primitives.rs [hunk 1]
```diff
@@ -58,7 +58,7 @@ fn sector_150(c: &mut Criterion) {
         Sector::with_center(
             BOUNDING_BOX.center(),
             BOUNDING_BOX.size.width,
-            -30.0.deg(),
+            -(30.0.deg()),
```
Comment (rfuest) [inline benches/primitives.rs:61]: Is this a clippy fix or why did you include this change in this PR?

File: src/image/sub_image.rs [hunk 2]
```diff
@@ -16,7 +16,7 @@ use crate::{
 ///
 /// [`sub_image`]: trait.ImageDrawableExt.html#tymethod.sub_image
 /// [module-level documentation]: super#sub-images
-#[derive(Debug)]
+#[derive(Debug, PartialEq, Eq)]
```
Comment (rfuest) [inline src/image/sub_image.rs:19]: I think we should also add `Clone` and `Copy` to the derives.

**Review by rfuest — CHANGES_REQUESTED** [review]
Looks good, but you should add a changelog entry for the added `Default` impls and for making it possible to compare `Image`s and `SubImage`s.

File: benches/primitives.rs (hunk 1 above)
Comment (za-songguo) [reply benches/primitives.rs:61]: > Is this a clippy fix or why did you include this change in this PR?

I ran `cargo clippy --fix` in the repo.

Comment (za-songguo) [reply benches/primitives.rs:61]: OK, I will add them to `CHANGELOG.md`.

File: src/image/sub_image.rs (hunk 2 above)
Comment (za-songguo) [reply src/image/sub_image.rs]: I have just committed the new changes to the repository.

**Review by rfuest — APPROVED** [review]
LGTM. Thanks for the PR.

**Pull request merged**